src/test/test-Transform.ts

```diff
@@ -0,0 +1,493 @@
+
+			expect(dest).to.deep.equal([[1, 2], [3]]);
+		});
+
+		it(
+			"applies backpressure to a `maxBatchSize` write from a `minBatchSize` write",
+			clockwise(async () => {
+				const batched = s.transform(batcher(2, { minBatchSize: 1 }));
+				const toWrite = [1, 2, 3];
+				const writes = [
+					...toWrite.map((n) => track(s.write(n))),
+					track(s.end()),
+				];
+
+				const resolvers: (() => void)[] = [];
+				batched.forEach(async (value) => {
+					const deferred = defer();
+					resolvers.push(deferred.resolve);
+					await deferred.promise;
+
+					batchResults.push(value);
+				});
+				await delay(1);
+				expect(batchResults).to.deep.equal([]);
+				expect(writes[0].isFulfilled).to.equal(true); // Commenced silently
+				expect(writes[1].isFulfilled).to.equal(true); // Added to the queue
+				expect(writes[2].isFulfilled).to.equal(false); // Triggers a batch write which can't begin yet
+				expect(resolvers[1]).to.equal(undefined);
+				resolvers[0]();
+				await delay(1);
+				expect(writes[2].isFulfilled).to.equal(false); // Batch write still can't finish, but...
+				expect(resolvers[1]).not.to.equal(undefined); // It's started!
+				resolvers[1]();
+				await delay(1);
+				expect(writes[2].isFulfilled).to.equal(true); // Now the batch write is finished...
+				expect(writes[3].isFulfilled).to.equal(true); // And the stream can end
+			})
+		);
+
+		it(
+			"forms batch not exceeding `maxBatchSize` if a batch write is pending",
+			clockwise(async () => {
+				const source = Stream.from([
+					{
+						value: 1,
+					},
+					{
+						value: 2,
+						workTime: 1,
+					},
+					{
+						value: 3,
+					},
+					{
+						value: 4,
+					},
+					{
+						value: 5,
+						workTime: 2,
+					},
+					{
+						value: 6,
+					},
+					{
+						value: 7,
+						wait: 3,
+					},
+					{
+						value: 8,
+					},
+				]);
+
+				const batched = pipeWithDelay(source).transform(
+					batcher(3, { minBatchSize: 2 })
+				);
+
+				const destAsync = resolveBatchToAsyncValues(batched);
+				const dest = await destAsync;
+
+				expect(dest).to.deep.equal([
+					[1, 2], //    Min batch size (this will take 1ms to write)
+					[3, 4, 5], // Max batch size streams in while first batch is writing (this will take 2ms to write)
+					[6, 7], //    Min batch size (7 arrives after a 3ms delay so the previous batch is processed and the new
+					//            batch is completed)
+					[8], //       Processed alone when stream ends
+				]);
+			})
+		);
+
+		it("bounces error", async () => {
+			const source = Stream.from([
+				{
+					value: 1,
+				},
+				{
+					value: 2,
+				},
+				{
+					value: 3,
+					throwError: true,
+				},
+			]);
+
+			const batched = pipeWithDelay(source).transform(batcher(3));
+
+			try {
+				await resolveBatchToAsyncValues(batched);
+				expect(true).to.equal(false);
+			} catch (e) {
+				expect(e).to.equal(boomError);
+			}
+		});
+
+		it("bounces error from `minBatchSize` batch", async () => {
```

Comment:
With 'bounces' I would expect that the original write would get it.

I think what's happening here is that the stream is aborted and its final result is tested for, correct?
As discussed, it would be best if we can indeed 'bounce' the error back, but if that's not feasible, we should change the wording to something like "aborts stream on error in `minBatchSize` batch".

Reply:
This now reflects the actual behaviour